react/components/EXPERIMENTAL_Table/hooks/useTableProportion.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react'
+import { Column } from '../index'
+
+export default function useTableProportion({ columns, ratio }: ProportionData) {
+  const calculatedWidths = useMemo(() => calculateWidths(columns, ratio), [
+    columns,
+    ratio,
+  ])
+
+  const sizedColumns = useMemo(
+    () => columns.map((col, i) => ({ ...col, width: calculatedWidths[i] })),
+    [columns, ratio]
+  )
+
+  return {
+    sizedColumns,
+  }
+}
+
+function calculateWidths(columns: Array<Column>, ratio: Array<number>) {
+  const slicedRatio = ratio.slice(0, columns.length)
+
+  const sum = (acc: number, num: number) => acc + num
+  const ratioSum = slicedRatio.reduce(sum, 0)
+  const minPortion = 100 / ratioSum
+
+  return slicedRatio.map(value => `${value * minPortion}%`)
```

Comment:
Why can not return number?

Reply:
This function is meant to return the array of percentages for each column. I represented a string so that I do not need to do it later since the manipulation is over.

Reply:
all right